content/actions/using-workflows/caching-dependencies-to-speed-up-workflows.md

```diff
@@ -369,7 +371,7 @@ jobs:
         env:
           GH_TOKEN: {% raw %}${{ secrets.GITHUB_TOKEN }}{% endraw %}
           REPO: {% raw %}${{ github.repository }}{% endraw %}
-          BRANCH: refs/pull/{% raw %}${{ github.event.pull_request.number }}{% endraw %}/merge
+          BRANCH: {% raw %}${{ github.ref_name }}{% endraw %}
```

Comment:
Indeed, using ref_name in workflow_dispatch is the correct approach.
![IMG_0161](https://github.com/github/docs/assets/69337624/15e927b4-3596-4e06-acf5-13ba00444e2c)
However, this approach is not correct in this example.
Because, [the cache is created for the merge ref (`refs/pull/.../merge`) when a cache is created by a workflow run triggered on a pull request.](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache) Therefore, it will not work correctly with ref_name.
![IMG_0162](https://github.com/github/docs/assets/69337624/07f0cc20-3671-428f-8352-6bd6c830e912)
![IMG_0163](https://github.com/github/docs/assets/69337624/d1007fc3-5a20-4abe-80cb-4be5c872d1a2)

Reply:
I think I might have been wrong - `ref_name` ends up being main/master after merge. We'd need `github.head_ref` here instead.